pkg/client/factory.go

```diff
@@ -111,52 +111,70 @@ func (p *Factory) AdminK8sInterface() (kubernetes.Interface, error) {
 	return kubernetes.NewForConfig(p.clientCfg)
 }
 
-func (p *Factory) DynamicClient(ctx *types.APIRequest) (dynamic.Interface, error) {
-	return newDynamicClient(ctx, p.clientCfg, p.impersonate)
+func (p *Factory) DynamicClient(ctx *types.APIRequest, warningHandler rest.WarningHandler) (dynamic.Interface, error) {
+	config := *p.clientCfg
+	config.WarningHandler = warningHandler
+	return newDynamicClient(ctx, &config, p.impersonate)
 }
 
-func (p *Factory) Client(ctx *types.APIRequest, s *types.APISchema, namespace string) (dynamic.ResourceInterface, error) {
-	return newClient(ctx, p.clientCfg, s, namespace, p.impersonate)
+func (p *Factory) Client(ctx *types.APIRequest, s *types.APISchema, namespace string, warningHandler rest.WarningHandler) (dynamic.ResourceInterface, error) {
+	config := *p.clientCfg
+	config.WarningHandler = warningHandler
```

Comment:
Could this just be added to `newClient` and/or `newDynamicClient`?

Reply:
> Could this just be added to `newClient` and/or `newDynamicClient`?

That might be a slightly cleaner way to go.

Reply:
Ok, I switched it up a bit and added it to newDynamicClient.  I think I like that a bit better.